QualityControl/lib/api.js

```diff
@@ -150,10 +150,9 @@ function errorHandler(err, res, status = 500) {
 function getTagsFromServices(services) {
   const prefix = model.queryPrefix;
   const tags = [];
-  for (const serviceName in services) {
-    if (services[serviceName] && services[serviceName].Tags && services[serviceName].Tags.length > 0) {
-      const tagsToBeAdded = services[serviceName].Tags;
-      tagsToBeAdded.filter((tag) => tag.startsWith(prefix))
+  for (const service in services) {
```

Comment:
I would rewrite a bit this part due to:
* now catalog has a simpler structure to use;
* we currently do not check for duplicates
```
const tagsSet = new Set(Object.values(services)
  .filter((service) => service?.length > 0)
  .flat());

const tags = [...tagsSet].map((tag) => ({name: tag}));

return tags
```
Or as one assignment only:
```
const tags = [...new Set(Object.values(services)
  .filter((service) => service?.length > 0)
  .flat())
].map((tag) => ({name: tag}));

return tags;
```

Reply:
I don't think this gonna work as you're filtering on tags not service names

Reply:
Are you referring to the one assignment only?
Basically what is happening is that we create a set which is flattened, containing only `tags` at this point. 
Out of that set we create an array and we filter the tags that do not start with QCPrefix

This was the initial requirement

Now that we have Nomad, we can filter on services as well but for that we do not have a standard naming convention from QC

Reply:
I was referring to `filter((tag) => tag?.startsWith(prefix))`, the prefix is for service names not for tags (see my proposed solution).

Reply:
Yes, but beforehand the prefix was filtering tags. I believe we should use both because if a user puts something else in a tag, we do not want to send it to QC as QC will only filter objects in the CCDB that start with the prefix. 
So, I believe the filter has to be there for tags

Reply:
Okay, indeed, I think you're right.